src/algebraic_topology/simplex_category.lean

```diff
@@ -402,4 +402,110 @@ full_subcategory_inclusion _
 
 end truncated
 
+section concrete
+
+instance : concrete_category.{0} simplex_category.{u} :=
+{ forget :=
+  { obj := λ i, fin (i.len + 1),
+    map := λ i j f, f.to_preorder_hom },
+  forget_faithful := {} }
+
+end concrete
+
+section epi_mono
+
+/-- A morphism in `simplex_category` is a monomorphism precisely when it is an injective function
+-/
+theorem mono_iff_injective {n m : simplex_category} {f : n ⟶ m} :
+  mono f ↔ function.injective f.to_preorder_hom :=
+begin
+  split,
+  { introsI m x y h,
+    have H : const n x ≫ f = const n y ≫ f,
+    { dsimp, rw h },
+    change (n.const x).to_preorder_hom 0 = (n.const y).to_preorder_hom 0,
+    rw cancel_mono f at H,
+    rw H },
+  { exact concrete_category.mono_of_injective f }
+end
+
+/-- A morphism in `simplex_category` is an epimorphism if and only if it is a surjective function
+-/
+lemma epi_iff_surjective {n m : simplex_category} {f: n ⟶ m} :
+  epi f ↔ function.surjective f.to_preorder_hom :=
+begin
+  split,
+  { intros hyp_f_epi x,
+    by_contradiction h_ab,
+    rw not_exists at h_ab,
+    -- The proof is by contradiction: assume f is not surjective,
+    -- then introduce two non-equal auxiliary functions equalizing f, and get a contradiction.
+    -- First we define the two auxiliary functions.
+    set chi_1 : m ⟶ [1] := hom.mk
+    ⟨λ u, if u ≤ x then 0 else 1,
+     by { intros a b a_leq_b,
+          by_cases h : a ≤ x,
+          { simp [h, fin.zero_le] },
+          { have b_ge_x : ¬(b ≤ x),
+            { contrapose! h,
+              exact a_leq_b.trans h },
+            simp [h, b_ge_x] }}⟩ with def_chi_1,
+    set chi_2 : m ⟶ [1] := hom.mk
+    ⟨λ u, if u < x then 0 else 1,
+     by { intros a b a_leq_b,
+          obtain rfl|a_lt_b := a_leq_b.eq_or_lt,
+          { refl },
+          by_cases h : a < x,
+          { simp [h, fin.zero_le] },
+          { have b_geq_x : ¬(b < x),
+            { contrapose! h,
+              exact a_lt_b.trans h },
+            simp [h, b_geq_x] }}⟩ with def_chi_2,
+    -- The two auxiliary functions equalize f
+    have f_comp_chi_i : f ≫ chi_1 = f ≫ chi_2,
+    { dsimp,
+      ext,
+      simp [le_iff_lt_or_eq, h_ab x_1] },
+    resetI,
```

Comment:
Is this needed to register the `epi` instance? If so, you can use `introsI` at the top of the proof.

Reply:
Yes, I'm using a `cancel_epi` at line 471, I'll do an `introsI` instead then.